src/GitVersionCore/FileLocking/Abstractions/IFileLocker.cs

```diff
@@ -0,0 +1,29 @@
+/*
+MIT License
+
+Copyright 2020 Teroneko
```

Comment:
@asbjornu not sure we have agreed on this, but do we allow to add custmo copyright to our code base?

Reply:
As I do agree with your license (that's why I enjoy to contribute to your library) the message I inserted could be shorten to:

```csharp
/* Copyright (c) 2020 Teroneko
 * Licensed under the MIT License. See LICENSE in the project root for license information.
 */
```

As I do claim copyright for this either way, it can be advisable to note its copright in the few files the copyright affects.

I really try to understand this nature of licensing and copyrights, but it is challenging me. Please let me know how you want to handle it. :)

Reply:
We don't have a strict copyright or licensing policy in GitVersion. It would probably have been clearer if we had a [CLA](https://en.wikipedia.org/wiki/Contributor_License_Agreement), but even without making this explicit, the copyright terms are such in most countries that copyright doesn't have to be explicitly stated – it is implicit through the publicaiton of copyrighted works.

So by you contributing these source files to GitVersion, you have copyright over them. But you also share this copyright with GitVersion, so we can both do whatever we wish with the files after they are contributed. We can rework them and change them to what's best for GitVersion, but you're also free to take the files and use them in another project of yours without asking anyone for permission.

So I don't think we need any explicit copyright or license information in our source files. No other files contain it so far and I think we should stay consistent. Thus I would kindly ask you to remove this, @teroneko.

Reply:
No problem. I was not sure about it. So I added it without knowledge. Now that you meantion it I know that copyright is always implicit. Thanks for your elaboration.